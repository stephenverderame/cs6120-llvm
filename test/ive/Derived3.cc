#include <cstdio>
int main()
{
    int LI = 100;
    for (int I = 0; I < 10; ++I) {
        int J = I * LI + LI;
        printf("%d\n", J);
    }
}