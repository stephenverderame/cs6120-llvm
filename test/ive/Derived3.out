Loop: 
1:                                                ; preds = %8, %0
  %.0 = phi i32 [ 0, %0 ], [ %9, %8 ]
  %2 = icmp slt i32 %.0, 10
  br i1 %2, label %4, label %3

Basic IV:   %.0 = phi i32 [ 0, %0 ], [ %9, %8 ]
Derived IV: %5 = %.0 * 100
Derived IV: %6 = %.0 * 100 + 100
Basic IV:   %9 = add nsw i32 %.0, 1
